libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -716,8 +717,21 @@ void _IotMqtt_ProcessKeepAlive( IotTaskPool_t pTaskPool,
         {
             IotLogDebug( "(MQTT connection %p) PINGRESP was received.", pMqttConnection );
 
-            /* PINGRESP was received. Schedule the next PINGREQ transmission. */
+            /* PINGRESP was received. Schedule the next PINGREQ transmission.
+             * Control actual delay with another variable, but keep this here
+             * so a PINGREQ will be transmitted in the next execution. */
             pMqttConnection->nextKeepAliveMs = pMqttConnection->keepAliveMs;
+
+            /* Check if keep alive time is nonzero and greater than wait time. */
+            if( pMqttConnection->keepAliveMs > IOT_MQTT_RESPONSE_WAIT_MS )
+            {
+                /* Subtract time taken for PINGRESP check. */
+                scheduleDelay = pMqttConnection->keepAliveMs - IOT_MQTT_RESPONSE_WAIT_MS;
+            }
+            else
```

Comment:
Do we need empty else marker for just if () is it only for if()  {} else if(){} scenario ?

Reply:
The other branches in this code also have empty else markers, even if there is not `else if`